src/gpu/intel/jit/gemm/selector/db/kernel.db

```diff
@@ -1162,6 +1162,15 @@ auto _CATALOG_ = kcatalog::toArray({
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 33, -1}, {-1, 48, -1}, {-1, 33, -1}, {-1, 48, -1}, {16, 16, 1}, "ABI"}, "at64x2 am32x2+m32@64 aB wg 4x1x4 ikr af k64 acb hi pt sr br sb64 sm sn sys nmk np", {16, (LoopType) 255, 128, {(LoopType) 209, (LoopType) 255, (LoopType) 2}, {262144, 262144, 16777216}, {262144, 262144, 16777216}, {16, 16, 64}, {4, 1, 4}, 1, (WGType) 1, 4101, 0, 4096, {16, 16, 2}, {true, true, true}}, {'E', 17, {1.00791e+06, 34809.3, 26236.6, 14043.4, 0, 0, 0.541503, 1.92574, 0.792396, 1.005, 0.012569, 0.0191402, 0.00806369, 1, 1.41743, 0.970564, 4.80381e-12}}},
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 49, -1}, {-1, 128, -1}, {-1, 49, -1}, {-1, 128, -1}, {16, 16, 1}, "ABI"}, "at64 am32+m32@32 aB wg 2x2x2 ikr xaf st rr acb hi pt sr br sb64 sn sys nmk np", {16, (LoopType) 255, 128, {(LoopType) 209, (LoopType) 255, (LoopType) 2}, {262144, 524288, 16777216}, {262144, 524288, 16777216}, {16, 32, 64}, {2, 2, 2}, 1, (WGType) 1, 4357, 0, 8192, {16, 16, 4}, {true, true, true}}, {'E', 17, {919791, 14572.7, 150898, 11874.8, 0, 0, 0.970933, 2.00496, 1.73249, 5.83627, 0.0164407, 0.0180319, 0.00303982, 0.139818, 1.26024, 1.04913, 1.75433e-12}}},
 {{'G', "gemm", {"N", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {1025, 129, 1}, {-1, 768, 3583}, {1025, 129, 1}, {-1, 768, 3583}, {16, 16, 1}, "ABIh"}, "at64x2+m64@8 am32x2+m32@8 aB wg 4x2 xaf st acb hi pt sr br sb64 sm sn grf256 sys np", {16, (LoopType) 255, 256, {(LoopType) 208, (LoopType) 255, (LoopType) 255}, {524288, 524288, 16777216}, {524288, 524288, 16777216}, {32, 32, 64}, {4, 2, 1}, 1, (WGType) 1, 257, 0, 0, {16, 16, 4}, {true, true, true}}, {'E', 17, {256058, 33684.4, 0, 0, 0, 0, 0.504291, 1.89712, 3.71959, 8.91844, 0.00785564, 0.00768242, 0.00317825, 1, 1.39291, 1.00693, 2.48102e-12}}},
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 8192}, {2048, 32, -1}, {-1, 1, 8192}, {2048, 32, -1}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x4 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 4}, 1, (WGType) 1, 261, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
+{{'G', "gemm", {"H", "H", "S"}, {"T", "N", "N"}}, {-1, -1, {-1, 1, 4096}, {2048, 32, 8191}, {-1, 1, 4096}, {2048, 32, 8191}, {16, 16, 1}, "IAB"}, "at32+m32@32 am64+m64@64 aB wg 4x1x2 vav sr br sb64 bm0 bk0 sys xaf sm sn st np grf256 ikr", {16, (LoopType) 255, 128, {(LoopType) 0, (LoopType) 1, (LoopType) 2}, {8192, 8192, 16777216}, {8192, 8192, 16777216}, {16, 16, 64}, {4, 1, 2}, 1, (WGType) 1, 261, 0, 2048, {16, 16, 4}, {true, true, true}}, {'W', 1, {128}}},
```

Comment:
Why are these two kernels fit using the W model, while the others are fit using the E model? With the restrictions it doesn't really matter, but that leads to my next question...

Have you tried measuring performance without accept/allow restrictions? If we can rely on the performance model instead, it will help with future optimization efforts.

Reply:
No, we don't use performance model. And we use shapes to distinguish the strategy so W/E part is just fictions. How could we express this properly in strategy?

Reply:
Okay, let me look into model E and see if it can capture the performance heuristics.